packages/components/src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,211 @@
+import clsx from "clsx";
+import React from "react";
+import { useUID } from "react-uid";
+import styles from "./Checkbox.module.css";
+
+// Allow all native HTML input attributes,
+// except for ones where we override type or strictness
+type CheckboxHTMLElementProps = Omit<
```

Comment:
thanks to @jeremiah-clothier for reminding me that this is probs more maintainable going forward. Note that we do lose some explicit documentation (i.e. `disabled`, `aria-label`, and `readOnly` props are no longer explicitly here). I think this is an okay trade-off because there inevitably may be more HTML attributes that people need (e.g. `readOnly` was added because of a React warning we saw), but open to push-back.

Another thing to note is Storybook is having issues parsing the props in either case 😬 

Now, it only shows props in `defaultArgs` or `argTypes` and doesn't load descriptions:
<img src="https://user-images.githubusercontent.com/15840841/149577811-07845ad2-90aa-48e1-86ad-e532531c43d4.png" width="400">

Previously, it loaded more props but didn't include `size` and marked `id` as required (which isn't true for the overall component):
<img src="https://user-images.githubusercontent.com/15840841/149582492-95b7b99b-7d28-4378-92bd-11ec61162ec2.png" width="600">

I'll make a ticket to dig into this more, maybe we're doing something wrong with the docgen config